pkg/provision/storage/commonStorage.go

```diff
@@ -71,8 +74,30 @@ func (p *CommonStorageProvisioner) ProvisionStorage(podAdditions *v1alpha1.PodAd
 	return nil
 }
 
-func (*CommonStorageProvisioner) CleanupWorkspaceStorage(workspace *dw.DevWorkspace, clusterAPI sync.ClusterAPI) error {
-	return runCommonPVCCleanupJob(workspace, clusterAPI)
+func (p *CommonStorageProvisioner) CleanupWorkspaceStorage(workspace *dw.DevWorkspace, clusterAPI sync.ClusterAPI) error {
+	_, totalWorkspaces, err := getSharedPVCWorkspaceCount(workspace.Namespace, clusterAPI)
+	if err != nil {
+		return err
+	}
+
+	// If the number of common + async workspaces that exist (started or stopped) is zero,
+	// delete common PVC instead of running cleanup job
+	if totalWorkspaces <= 1 {
+		existingPVC := &corev1.PersistentVolumeClaim{}
+		namespacedName := types.NamespacedName{Name: config.Workspace.PVCName, Namespace: workspace.Namespace}
+		err := clusterAPI.Client.Get(clusterAPI.Ctx, namespacedName, existingPVC)
+		if err != nil {
```

Comment:
I'm not so sure about the error handling logic here. Currently, if a `k8sErrors.IsNotFound` occurs then we will still return the error. Maybe if the PVC can't be found, we just assume the PVC is already deleted (in other words, we should ignore `k8sErrors.IsNotFound`)?

Reply:
It's important to handle this correctly as there might be two in-flight reconciles that attempt to delete the PVC at the same time. Generally, for `client.Get` requests, we treat `k8sErrors.IsNotFound` as "assume object is already deleted", and propagate other errors. In this case, this would mean returning `nil`, as we don't need to call `client.Delete` below.